src/include/distributed/shard_cleaner.h

```diff
@@ -17,8 +17,90 @@ extern bool DeferShardDeleteOnMove;
 extern double DesiredPercentFreeAfterMove;
 extern bool CheckAvailableSpaceBeforeMove;
 
+extern bool DeferShardDeleteOnSplit;
+extern int NextOperationId;
+extern int NextCleanupRecordId;
+
 extern int TryDropOrphanedShards(bool waitForLocks);
-extern int DropOrphanedShards(bool waitForLocks);
 extern void DropOrphanedShardsInSeparateTransaction(void);
 
+/* Members for cleanup infrastructure */
+typedef uint64 OperationId;
+extern OperationId CurrentOperationId;
+
+/*
+ * CleanupResource represents the Resource type in cleanup records.
+ */
+typedef enum CleanupObject
+{
+	CLEANUP_INVALID = 0,
+	CLEANUP_SHARD_PLACEMENT = 1
+} CleanupObject;
+
+/*
+ * CleanupPolicy represents the policy type for cleanup records.
+ */
+typedef enum CleanupPolicy
+{
+	/*
+	 * Resources that are transient and always need clean up after the operation is completed.
+	 * (Example: Dummy Shards for Non-Blocking splits)
+	 */
+	CLEANUP_ALWAYS = 0,
```

Comment:
would be nice to use different prefixes between CleanupObject and CleanupPolicy

Reply:
Fixed.